pkg/obfuscator/mac_address.go

```diff
@@ -29,14 +46,23 @@ func (m *macAddressObfuscator) Report() map[string]string {
 	return m.ReplacementTracker.Report()
 }
 
-func NewMacAddressObfuscator() Obfuscator {
+func NewMacAddressObfuscator(replacementType schema.ObfuscateReplacementType) (Obfuscator, error) {
+	if replacementType != schema.ObfuscateReplacementTypeStatic && replacementType != schema.ObfuscateReplacementTypeConsistent {
+		return nil, fmt.Errorf("unsupported replacement type: %s", replacementType)
+	}
 	// this regex differs from the standard `(?:[0-9a-fA-F]([:-])?){12}`, to not match very frequently happening UUIDs in K8s
 	// the main culprit is the support for squashed MACs like '69806FE67C05', which won't be supported with the below
 	regex := regexp.MustCompile(`([0-9a-fA-F]{2}[:-]){5}[0-9a-fA-F]{2}`)
 
 	reporter := NewSimpleTracker()
+	generator := obfsGenerator{
```

Comment:
can you wrap this with a New* function as well? that way you can decouple the struct and its interface

Reply:
True, addressed